### ISSUE TITLE
Crash while interacting with various RFTools machines

### ISSUE THREAD

**Issue by tedmihu** [McJtyMods/RFTools#1813]
Issue description: 
Not consistently, and not always on the same action, but when placing certain machines, or opening their GUI, client experiences a hard crash..

Steps to reproduce: 
I can't offer reproduction steps because the issue is not present constantly. Sometimes it's after placing a builder (try 100% and 0% infused). A lot of times it's when opening the crafter GUI (try tier 1, 2, 3) that's been the most consistent one, but it isn't every time, only sometimes.

Versions:
* Minecraft: 
1.12.2
* Forge: 
14.23.4.2708
* CompatLayer (only if on Minecraft 1.10 or 1.11):
N/A
* McJtyLib:
3.0.2
* RFTools:
7.54

Possibly incompatible mods and versions: 
I use optifine, but I am posting crash logs from two other users who don't use optifine.

Relevant logs, if any: 

User PhantomSyndrome, no optifine - https://hastebin.com/wehujunifi.coffeescript
User UncleQball, no optifine - https://hastebin.com/rumipofuxo.sql
Me, with optifine 
- https://hastebin.com/icepasedoh.coffeescript
- https://hastebin.com/afalajabur.sql

**Comment by Grygon** [comment]
I just got this crash too, [here's my log](https://paste.ee/p/61JUY)

Going to try the new mcjtylib and see if it helps.

**Comment by josephcsible** [comment]
That crash is completely different from this one. Please open a new issue for it, and also see if you can reproduce it without Mouse Tweaks.

**Comment by Grygon** [comment]
Whoops, so it is, I saw a similar enough thing and figured it was the same issue. The new lib did seem to fix it though, so I'm thinking they're probably related.

**Comment by tedmihu** [comment]
Using the mcjtylib-1.12-3.0.2+f2de7c9 jar server-side only, using optifine, all version info is the same as the first post in this thread. 

https://hastebin.com/jobazoteli.sql

**Comment by josephcsible** [comment]
Unfortunately, using the jar on the server doesn't make the client's crash log useful.

**Comment by tedmihu** [comment]
My server dev just told me that. Thought I'd post it just in case. Any chance you can release this jar on curseforge marked as alpha? Would love to update my pack even with the understanding that it's an unstable release.

**Comment by tedmihu** [comment]
It didn't crash, and it only happened the first time I clicked on a crafter, but when opening a crafter, I saw my items and nothing else, like a fully transparent GUI.

**Comment by josephcsible** [comment]
Later tonight, I'll make another jar that will help get to the bottom of this.

**Comment by josephcsible** [comment]
Can you try this build? It should definitely give you a useful crash report. https://github.com/josephcsible/McJtyLib/releases/download/1.12-3.0.2%2Bblamingnonnulllist/mcjtylib-1.12-3.0.2+blamingnonnulllist.jar

**Comment by wormzjl** [comment]
Got a similar crash opening the coal gen gui
https://gist.github.com/wormzjl/27e05284da8343dece09ed596e022491

**Comment by josephcsible** [comment]
Can you reproduce it with either of the test builds in my last post?

**Comment by tedmihu** [comment]
Forge 2732
McJty 3.303
With optifine

Everything else is the same.

**Comment by CritFlaw** [comment]
Ok, so I haven't done _extensive_ testing without NEI, but it does seem that reducing that mod has resulted in no more crash. That being said, is this a bug I should report to to NEI? Because it's a really useful client-side mod for filtering items when searching, and it also includes light overlays.

**Comment by josephcsible** [comment]
Yes, report it to them. Also, it looks like everything that NEI does can be done by some combination of other mods: https://docs.google.com/spreadsheets/d/1NEWZjKvWB-SibY1BXc1zoYkK7AhjMMBZess9wMy--6w/htmlview#gid=0

**Comment by CritFlaw** [comment]
Oh my god someone actually made a spreadsheet for ths? :')

**Comment by CritFlaw** [comment]
Unfortunately one of the core features I'm looking for, double clicking the search bar to black out all items except the ones you're searching for, doesn't seem to be replaceable.

**Comment by MauveCloud** [comment]
> Unfortunately one of the core features I'm looking for, double clicking the search bar to black out all items except the ones you're searching for, doesn't seem to be replaceable.

If you're searching in chests the mod "Quark" has its own search box (independent of JEI/NEI afaik) that does that, though I'll admit it is more limited (seems to only search by names, not mods or ore dictionary)
